core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java

```diff
@@ -25,6 +25,7 @@ public final class ReflectiveClassBuildItem extends MultiBuildItem {
     private final boolean weak;
     private final boolean serialization;
     private final boolean unsafeAllocated;
+    private final String reason;
```

Comment:
Isn't this an array? At least the JSON example says it so

Reply:
It's `ReflectiveClassBuildItem` has a single reason that it was created for, but you might have multiple `ReflectiveClassBuildItem`s that were generated in different places for different reasons and refer to the same class. In this case you will get an array of reasons for this class registration. The relevant code is in https://github.com/quarkusio/quarkus/pull/42205/files#diff-3cb159eef1e3a292b96e837164047fe8be00767d4e8c2fba80d846ecdf4b8253